ntp/protocol/ntp.go

```diff
@@ -42,28 +44,25 @@ func Unix(seconds, fractions uint32) time.Time {
 	return time.Unix(secs, nanos)
 }
 
-// abs returns the absolute value of x
-func abs(x int64) int64 {
-	if x < 0 {
-		return -x
-	}
-	return x
+// Offset uses NTP algorithm for clock offset
+func Offset(originTime, serverReceiveTime, serverTransmitTime, clientReceiveTime time.Time) int64 {
+	outboundClockDelta := serverReceiveTime.Sub(originTime).Nanoseconds()
+	inboundClockDelta := serverTransmitTime.Sub(clientReceiveTime).Nanoseconds()
+
+	return (outboundClockDelta + inboundClockDelta) / 2
 }
 
-// AvgNetworkDelay uses formula from RFC #958 to calculate average network delay
-func AvgNetworkDelay(clientTransmitTime, serverReceiveTime, serverTransmitTime, clientReceiveTime time.Time) int64 {
-	forwardPath := serverReceiveTime.Sub(clientTransmitTime).Nanoseconds()
-	returnPath := clientReceiveTime.Sub(serverTransmitTime).Nanoseconds()
+// RoundTripDelay uses NTP algorithm for roundtrip network delay
+func RoundTripDelay(originTime, serverReceiveTime, serverTransmitTime, clientReceiveTime time.Time) int64 {
```

Comment:
I don't follow this change. The behaviour is completely changing (calculation). How are we simply swapping "formula from RFC #958" to a different calculations?

Reply:
This the Delay formula from the RFCs. ` d = (t4 - t1) - (t3 - t2) `

Reply:
It's still not clear. In the past we were calculating AvgNetworkDelay. Which has to be positive (and average). Maybe I am missing something but I don't see how can you replace it with possibly negative and definitely not average...